src/features/recovery/RecoveryDialogs.tsx

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+    addConfirmBeforeClose,
+    Alert,
+    Button,
+    clearConfirmBeforeClose,
+    ConfirmationDialog,
+    DialogButton,
+    GenericDialog,
+    logger,
+    useStopwatch,
+} from '@nordicsemiconductor/pc-nrfconnect-shared';
+
+import {
+    isSessionRecoveryPending,
+    setSavePending,
+} from '../../slices/appSlice';
+import { formatDuration, formatTimestamp } from '../../utils/formatters';
+import TimeComponent from '../ProgressDialog/TimeComponent';
+import { RecoveryManager } from './RecoveryManager';
+import {
+    DeleteAllSessions,
+    RemoveSessionByFilePath,
+    Session,
+} from './SessionsListFileHandler';
+
+const SessionItem = ({
+    session,
+    onRecoverClick,
+    onRemoveClick,
+}: {
+    session: Session;
+    onRecoverClick: (session: Session) => void;
+    onRemoveClick: (session: Session) => void;
+}) => (
+    <div className="tw-flex tw-flex-col tw-justify-end tw-gap-11 tw-bg-gray-800 tw-p-3 tw-text-white">
+        <div className="tw-flex tw-flex-grow tw-flex-col tw-justify-between tw-gap-2">
+            <div className="tw-flex tw-flex-row tw-justify-between">
+                <div>
+                    <div className="tw-text-xs">Start time</div>
+                    <div>{formatTimestamp(session.startTime)}</div>
+                </div>
+                <div>
+                    <div className="tw-text-xs">Duration</div>
+                    <div>{formatDuration(session.samplingDuration || 0)}</div>
+                </div>
+                <div>
+                    <div className="tw-text-xs">Sampling rate</div>
+                    <div>{session.samplingRate}</div>
+                </div>
+                <div className="tw-content-center tw-align-middle">
+                    <div className="tw-flex tw-flex-row tw-gap-2">
+                        <Button
+                            variant="secondary"
+                            onClick={() => onRemoveClick(session)}
+                        >
+                            Delete
+                        </Button>
+                        <Button
+                            variant="primary"
+                            className="tw-w-[60px]"
+                            onClick={() => onRecoverClick(session)}
+                        >
+                            {session.alreadyRecovered ? 'Load' : 'Recover'}
```

Comment:
Why do we have two options for recovery? As a user, I don't care if I recovered a session before and did not save so I need to load or if I haven't yet recovered a session. Given that both are about unsaved data, why not just have one option -- "Recover"? That also makes it easier to communicate it around it by avoiding adding notifications, two versions of preceding paragraph, and info in the docs.